backend/app/mcp_server/tool_registry.py

```
@@ -0,0 +1,289 @@
+    else:
+        return json.dumps(result, ensure_ascii=False, default=str)
+
+
+def _set_function_signature(func: Callable, params: List[Dict[str, Any]]) -> None:
+    """Set function signature for FastMCP parameter inference.
+
+    Args:
+        func: Function to modify
+        params: List of parameter definitions
+    """
+    from typing import Optional
```

Ruff (0.14.14)

[error] Redefinition of unused `Optional` from line 15: `Optional` redefined here

Remove definition: `Optional`

(F811)